cellprofiler/modules/colortogray.py

```diff
@@ -208,30 +208,22 @@ def create_settings(self):
 
     channel_names = ["Red: 1", "Green: 2", "Blue: 3", "Alpha: 4"]
 
-    def _get_next_channel(self):
-        while True:
-            channel = str(len(self.channel_names) + 1)
-            yield channel
-
     def add_channel(self, can_remove=True):
         '''Add another channel to the channels list'''
         group = cps.SettingsGroup()
         group.can_remove = can_remove
-        if len(self.channels) == len(self.channel_names):
-            ch = next(self._get_next_channel())
-            self.channel_names.append(ch)
-        group.append("channel_choice", cps.Choice(
+        group.append("channel_choice", cps.Integer(
             text="Channel number",
-            choices=self.channel_names,
-            value=self.channel_names[len(self.channels)],
+            value=len(self.channels)+1,
```

Comment:
Add a space between operators, e.g.

```Python
value=len(self.channels) + 1
```

Reply:
Thanks for the hint! I will check the formatting once more.